### PR TITLE
262 feature request add formula a2 from nen en 1993 1 9 c2 2012

### LINKED ISSUE #262: [✨ Feature request]: Add formula A.2 from NEN_EN_1993_1_9_C2_2012
### Description

Add formula A.1 from NEN_EN_1993_1_9_C2_2012

### Acceptance Criteria

- [x] Add the formula to the code
- [x] Add latex expression to the formula
- [x] Add unittests
- [x] Add docstrings
- [x] Update table in docs with ✔️

### Dependencies

_No response_

### Confirmation

- [X] I have checked that this feature does not already exist in the issue tracker.
- [X] I have read and understood the contribution guidelines.

### PR DESCRIPTION
**Pull request by GerjanDorgelo**
## Description

Formula A.2 from 1993-1-9+C2:2012 was added.

Fixes #262 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: blueprints/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/formula_a_2.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        -------
+        None
+        """
+        super().__init__()
+        self.d_d = d_d
+
+    @staticmethod
+    def _evaluate(
+        d_d: DIMENSIONLESS,
+    ) -> bool:
```
Comment (Simone-de-Rijke) [inline blueprints/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/formula_a_2.py:39]: Geen Bool maar string gebruiken. 'OK' of 'NOT OK'. Of een Enumerate of een dictionary

File: tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+"""Testing formula A.1 from NEN-EN 1993-1-9+C2:2012: Annex A - Determination of fatigue load parameters and verification formats."""
+
+import pytest
+
+from blueprints.codes.eurocode.nen_en_1993_1_9_c2_2012.annex_a_determination_of_fatigue_load_parameters_and_verification_formats.formula_a_2 import (
+    FormADot2CriteriaBasedOnDamageAccumulation,
```
Comment (Simone-de-Rijke) [inline tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:6]: Import is sorted in a way you can do multiple function imports at once. in this case this is not neccesary. The "(" "," and ")" can be removed.

File: tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    """Validation for formula A.2 from NEN-EN 1993-1-9+C2:2012."""
+
+    def test_evaluation_if_ok(self) -> None:
+        """Test the evaluation of the result."""
+        # example values
+        d_d = 0.8  # -
+
+        form_a_1 = FormADot2CriteriaBasedOnDamageAccumulation(d_d=d_d)
+        # manually calculated result
+        manually_calculated_result = True
```
Comment (Simone-de-Rijke) [inline tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:21]: This changes when changing from boolean to other format.

File: tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        assert form_a_1 == pytest.approx(expected=manually_calculated_result, rel=1e-9)
+
+    def test_evaluation_if_not_ok(self) -> None:
+        """Test the evaluation of the result."""
+        # example values
+        d_d = 1.1  # -
+
+        form_a_1 = FormADot2CriteriaBasedOnDamageAccumulation(d_d=d_d)
+        # manually calculated result
+        manually_calculated_result = False
```
Comment (Simone-de-Rijke) [inline tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:32]: this too

File: tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+        assert form_a_1 == pytest.approx(expected=manually_calculated_result, rel=1e-9)
+
+    def test_raise_error_if_negative_n_e(self) -> None:
+        """Test that a NegativeValueError is raised when a negative value is passed for n_E."""
+        d_d = -0.1  # -
```
Comment (Simone-de-Rijke) [inline tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:38]: def test_raise_error_if_negative_d_d(self, d_d: float) ->None:
"""Test that a NegativeValueError is raised when a negative value is passed."""

Comment (Simone-de-Rijke) [reply tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:38]: de d_d=-0.1 is dan niet benodigd. zie ook [https://github.com/Blueprints-org/blueprints/blob/main/tests/codes/eurocode/nen_en_1993_1_1_c2_a1_2016/chapter_6_ultimate_limit_state/test_formula_6_5.py](url)

File: tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+from blueprints.validations import NegativeValueError
+
+
+class TestFormADot2CriteriaBasedOnDamageAccumulation:
+    """Validation for formula A.2 from NEN-EN 1993-1-9+C2:2012."""
+
+    def test_evaluation_if_ok(self) -> None:
+        """Test the evaluation of the result."""
+        # example values
+        d_d = 0.8  # -
```
Comment (Simone-de-Rijke) [inline tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:17]: Je kan een psotieve en negatieve in 1x parametrisen

File: tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+from blueprints.codes.eurocode.nen_en_1993_1_9_c2_2012.annex_a_determination_of_fatigue_load_parameters_and_verification_formats.formula_a_2 import (
+    FormADot2CriteriaBasedOnDamageAccumulation,
+)
+from blueprints.validations import NegativeValueError
+
+
+class TestFormADot2CriteriaBasedOnDamageAccumulation:
+    """Validation for formula A.2 from NEN-EN 1993-1-9+C2:2012."""
+
+    def test_evaluation_if_ok(self) -> None:
```
Comment (Simone-de-Rijke) [inline tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:14]:
```suggestion
 @pytest.mark.parametrize(
        ("d_d", "resman"),
        [
            (0.8, True),
            (1.1, False),
        ],
    )
    def test_evaluation_if_ok(self, d_d: float, resman: bool) -> None:
```

File: blueprints/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/formula_a_2.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    """Class representing formula A.2 for the calculation of the damage during the design life."""
+
+    label = "A.2"
+    source_document = NEN_EN_1993_1_9_C2_2012
+
+    def __init__(
+        self,
+        d_d: DIMENSIONLESS,
+    ) -> None:
+        """[:math:`OK?`] Criteria met, based on damage accumulation.
```
[the remark this answers is not in the document]
Comment (GerjanDorgelo) [reply blueprints/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/formula_a_2.py:20]: I changed it to OK / NOT OK after discussing review with Simone

File: blueprints/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/formula_a_2.py (hunk 1 above)
Comment (GerjanDorgelo) [reply blueprints/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/formula_a_2.py:39]: Is not possible in current architecture. Proposal to keep format

File: tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py (hunk 2 above)
Comment (GerjanDorgelo) [reply tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:6]: Is forced by Black extention. Proposal to keep format.

File: tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py (hunk 3 above)
Comment (GerjanDorgelo) [reply tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:21]: Not relevant after comment above.

File: tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py (hunk 4 above)
Comment (GerjanDorgelo) [reply tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:32]: Not relevant after comment above.

File: tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py (hunk 5 above)
Comment (GerjanDorgelo) [reply tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:38]: would require the  @pytest.mark.parametrize, proposal to keep the way it is.

File: tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py (hunk 7 above)
Comment (GerjanDorgelo) [reply tests/codes/eurocode/nen_en_1993_1_9_c2_2012/annex_a_determination_of_fatigue_load_parameters_and_verification_formats/test_formula_a_2.py:14]: Good suggestion, implemented!

**Pull request merged**